src/main/java/com/iota/iri/service/API.java

```diff
@@ -110,7 +110,7 @@ public API(Iota instance, IXI ixi) {
     }
 
     public void init() throws IOException {
-        readPreviousEpochsSpentAddresses();
+        readPreviousEpochsSpentAddresses(instance.configuration.booling(DefaultConfSettings.TESTNET));
```

Comment:
why not use `testNet` field initialized in the constructor?

Reply:
fixed

Reply:
so the reason to pass it as an argument is so it's explicit that `readPreviousEpochsSpentAddresses` depends on `testNet`? because this can also be accessed from within the method itself.

Reply:
It makes it clear for the reader of `init()` that `readPreviousEpochsSpentAddresses` depends on the value of testnet. Like the `if` you had in your closed PR.

I did it this way automatically because if this method will be called again from another place in `API`, no one will forget that it depends on testnet.

You can rebut and say that this method will never be called in `API` more than once (which makes sense). In this case the way you did it in the closed PR was preferable.